### ISSUE TITLE
Google oauth login issue

### ISSUE THREAD

**Issue by exonian** [PeteAUK/warscroll-designer#11]
**Describe the bug**
When trying to log in with Google, after choosing my account and getting redirected out, I get a blank page at "https://runebrush.pa-sy.com/warscroll/oauth/google/", with HTTP 500, and therefore the login fails

**To Reproduce**
Use the "sign in with google" menu option and choose a google account

**Expected behaviour**
Successful login and redirect back to the site

**Desktop (please complete the following information):**
 - OS: Windows 10 Home
 - Browser: Chrome 112.0.5615.138, Firefox 112.0.2 (64-bit) (with my usual extensions but again in private browsing mode)

**Comment by exonian** [comment]
Also reproduced on mobile (Firefox on Android)

**Comment by PeteAUK** [comment]
Hi.  Could you double check you're not using the archived old version of the tool.  The correct URL is: https://runebrush.pa-sy.com/warscroll/

**Comment by exonian** [comment]
![image](https://user-images.githubusercontent.com/1237986/236648747-b5866d88-ff2a-490e-badf-b62e17f88a04.png)

**Comment by exonian** [comment]
From the dev tools console, on page load I see the following error:
`[GSI_LOGGER]: The given origin is not allowed for the given client ID.`
and an HTTP 403 GET request to `https://accounts.google.com/gsi/button?<various params>` that seem relevant and a couple of warnings (cookie samesite attribute and content security policy) that I'm less sure about

**Comment by PeteAUK** [comment]
Thanks for the info, I'll have a look into this.  Basically the address that's throwing the 500 error no longer exists as it's for the old Google Login script which is why I double checked you were on the latest version.  My PC & phone are both logging in fine with various Google accounts, which is suggestive something else is going on.

Out of interest have you logged in previously or is this first time?

**Comment by exonian** [comment]
Oh I see!
Never logged in previously on either version

**Comment by PeteAUK** [comment]
Ah, have found the issue.  It was partly related to Google Login and partly related to me.  Previously Google Login would provide a "locale" as part of the login process, which I recorded as part of the user registration (my original thinking was that I could always create different language versions of the editor).  This data isn't provided and there wasn't a default value for this field in the database (in my defence the database structure was created back in 2016!).  What that meant was for people such as yourself who had never logged in previously a new user wasn't being created.

The short version is this should now work for you without any further issues - thanks for creating the issue, this could have gone on for months otherwise!

**Comment by exonian** [comment]
It has worked! Thank you! I obsessively retried it half a minute ago not having looked here and was pleasantly surprised it worked!

Thank you so much for fixing it, I'm really excited to use it more. I'm glad my report was useful, and thanks for the extra bit of explanation as that's always interesting :)
